### PR TITLE
[wand-watcher] Add "no use" rooms, debugging, and update base.yml

### PR DESCRIPTION
**Pull request by KatoakDR**
Thanks for the wand-watcher script. I've been using mefwatch for a while and had just updated my local copy to handle rooms where the wand won't work and to ignore those so it doesn't repeatedly whip the wands out only to keep finding out I'm standing in the bank and its pointless to do so.

I then saw your pull request https://github.com/rpherbig/dr-scripts/pull/4418 and thought you might be open to these additions. Thanks for your consideration!

---

- The `base.yaml` changes is just bringing it up to date with what's been merged to master.
- Also adds new configuration option `wand_watcher_no_use_rooms`, a list of words to match in room titles or the actual room ids of places to not try and use the wands because they won't work there (e.g. like bank window).
- To help with testing, adding `debug` argument and various `echo` statements.
- Pretty sure my code editor trimmed trailing whitespace, the diff of the script looks nicer with "ignore whitespace" on

### CODE REVIEW

File: profiles/base.yaml [hunk 1]
```diff
@@ -1447,5 +1453,10 @@ wand_watcher_no_use_scripts:  # Don't try to grab any wands while these scripts
   - burgle
   - go2
 
+wand_watcher_no_use_rooms:  # Don't try to grab any wands while in these rooms.
+  - Vault                   # Such as rooms that give you the "Something in the area is interfering with the magical device." message.
+  - Bank                    # You can specify room titles and room ids.
+  - 1900
```
Comment (KatoakDR) [inline profiles/base.yaml:1459]: Lines 1456-1459 are my additions; the rest are just bringing this file up to date with https://github.com/rpherbig/dr-scripts/commit/2aaa0dbc29ea78706291395a2869b09a40154972

File: wand-watcher.lic [hunk 2]
```diff
@@ -12,18 +12,26 @@ class WandWatcher
...
     @passive_delay = settings.wand_watcher_passive_delay || 60
     @activation_failure_messages = [/^The .* remains inert/]
+    @activation_blocked_messages = ["You shouldn't disrupt the area", "Something in the area is interfering"]
+    @debug = UserVars.wand_watcher_debug || args.debug || false
     UserVars.wand_watcher_timers = {} unless UserVars.wand_watcher_timers
 
     #don't allow users to make foolish mistakes, always prevent use during go2/burgle
-    @no_use_scripts << 'go2'
-    @no_use_scripts << 'burgle'
+    ['go2', 'burgle'].each { |script| @no_use_scripts.push(script) unless @no_use_scripts.include?(script) }
```
Comment (KatoakDR) [inline wand-watcher.lic:34]: If 'go2' or 'burgle' are already in the user's `no_use_scripts` yaml config, then the original lines 25-26 -- although very succinct -- would duplicate those entries. This tweak is my novice ruby mind's way of only adding the entries if they don't already exist.

**Comment by vtcifer** [comment]
You're more than welcome to make comments on PRs on the main dr-scripts repo, but I keep this as a personal one and don't merge requests from others.

**Comment by KatoakDR** [comment]
@vtcifer Will do, thanks, and hopefully this didn't come across as stepping on your toes so to speak!

**Pull request closed without merging**